front/app/utils/dateUtils.ts

```diff
@@ -29,6 +31,26 @@ type RelativeTimeFormatUnit =
   | 'second'
   | 'seconds';
 
+// this function calculates whether there is any overlap between two time ranges.
+// Ref: https://stackoverflow.com/a/74363870
```

Comment:
Will actually be removed once the BE can support some new date filtering. So this is just temporary while I wait for that :)